karb/parser.js

```diff
@@ -51,14 +57,19 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
     // /document/print?ids=KLI-KCC-1103101-n&title=PDF
     // /document/print?ids=KLI-KCC-1103101-n&title=PDF
     // /document/print?ids=KLI-KA-Paulsson-2016-Ch02%2CKLI-KA-Paulsson-2016-Ch03%2CKLI-KA-Paulsson-2016-Ch04%2CKLI-KA-Paulsson-2016-Ch05&pdf=
-
-    result.rtype  = 'BOOK_CHAPTERS_BUNDLE';
     result.mime   = 'PDF';
 
-    if (!Object.prototype.hasOwnProperty.call(param, 'pdf')) {
-      result.rtype  = undefined;
+    if ((param.id && param.id.includes('Ch')) || (param.ids && param.ids.includes('Ch'))) {
+      if ((param.id && param.id.match(/Ch/g).length > 1) || (param.ids && param.ids.match(/Ch/g).length > 1)) {
+        result.rtype  = 'BOOK_CHAPTERS_BUNDLE';
+      } else {
+        result.rtype  = 'BOOK_SECTION';
+      }
+    } else if (param.id || param.ids) {
+      result.rtype  = 'ARTICLE';
       result.unitid = param.ids;
```

Comment:
According to your condition, the unitid should be ``result.unitid = param.id || param.ids``